### PR TITLE
Allow running pipeline against specific PR

### PR DESCRIPTION
**Pull request by sutaakar**

### CODE REVIEW

**Review by radtriste — APPROVED** [review]
lgtm

File: kogito-cloud-operator-build.jenkinsfile [hunk 1]
```diff
@@ -30,7 +30,19 @@ pipeline {
...
+        stage("Checkout pull request branch"){
+            when {
+                expression {
+                    return env.PULL_REQUEST != '';
+                }
+            }
+            steps {
+                dir("${WORKING_DIR}") {
+                    sh "git fetch origin pull/${PULL_REQUEST}/head:tempbranch"
+                    sh "git checkout tempbranch"
```
Comment (radtriste) [inline kogito-cloud-operator-build.jenkinsfile:42]: maybe update the build name with the `PULL_REQUEST` name ?

Comment (sutaakar) [reply kogito-cloud-operator-build.jenkinsfile:42]: Yes, good point.

Comment (MarianMacik) [reply kogito-cloud-operator-build.jenkinsfile:42]: +1 that would be nice

**Review by MarianMacik — APPROVED** [review]

**Comment by sutaakar** [comment]
Updated, now the build name is stored in format PR-<number> if the PR build is used.

**Review by radtriste — APPROVED** [review]

**Pull request merged**